core-java-modules/core-java-string-operations-7/src/test/java/com/baeldung/stringcontainscurrencysymbol/CurrencySymbolFinderTest.java

```diff
@@ -0,0 +1,37 @@
+package com.baeldung.stringcontainscurrencysymbol;
+
+import org.junit.Test;
+
+import java.util.Locale;
+
+import static com.baeldung.stringcontainscurrencysymbol.CurrencySymbolFinder.*;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class CurrencySymbolFinderTest {
+
+    private static final String SINGLE_SIGN_REGEX = "\\p{Sc}";
+    private static final String MULTIPLE_SIGNS_REGEX = "^\\p{Sc}{3}[1-9]";
```

Comment:
Why is this not just:
```suggestion
    private static final String MULTIPLE_SIGNS_REGEX = "^\\p{Sc}{3}";
```

Reply:
Without indicating that after the currency sign we expect some digits test passes whenever we put three or more signs like this:
String poundValue = "££££4,098.09";
as well as
String poundValue = "£££4,098.09";

I agree with your comment that it would not be real world scenario. Rather we can check if there is exactly one occurance of currency symbol.